File: leetcode/Python/add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        i, j, c, s = len(a)-1, len(b)-1, 0, 0
        sol = ''
        while i>=0 and j>=0:
            s = int(a[i])+int(b[j])+c
            c = s//2
            s%=2
            sol+='1' if s>0 else '0'
            i, j = i-1, j-1
        while i>=0:
            s = int(a[i])+c
            c = s//2
            s%=2
            sol+='1' if s>0 else '0'
            i = i-1
        while j>=0:
            s = int(b[j])+c
            c = s//2
            s%=2
            sol+='1' if s>0 else '0'
            j = j-1
        if c > 0:
            sol += '1'
        return sol[::-1]

File: leetcode/C++/add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string sol;
        int i=a.length()-1,j=b.length()-1,c=0;
        for(; i>=0 && j>=0; --i,--j){
            sol.push_back((a[i]+b[j]-'0'-'0'+c)%2+'0');
            c=(a[i]+b[j]-'0'-'0'+c)/2;
        }
        for(; i>=0; --i){
            sol.push_back((a[i]-'0'+c)%2+'0');
            c=(a[i]-'0'+c)/2;
        }
        for(; j>=0; --j){
            sol.push_back((b[j]-'0'+c)%2+'0');
            c=(b[j]-'0'+c)/2;
        }
        if(c) sol.push_back('1');
        reverse(sol.begin(), sol.end());
        return sol;
    }
};